### ISSUE TITLE
Unable to build from source

### ISSUE THREAD

**Issue by atorkhov** [racket/racket#1144]
When building racket from source distribution `racket-6.2.1-src.tgz`, I'm getting the following errors:

```
raco setup: --- summary of errors ---
raco setup: error: during building docs for <pkgs>/gui-doc/scribblings/framework/framework.scrbl
raco setup:   query-rows: attempt to write a readonly database
raco setup:     error code: 776
raco setup:     SQL: "SELECT pathid FROM documented WHERE stag=$1"
raco setup:     database: #<path:/home/alex/tmp/racket/usr/share/doc/racket/docindex.sqlite>
raco setup:     mode: 'read-only
raco setup:     file permissions: (write read)
raco setup:     context...:
raco setup:      /home/alex/tmp/racket/usr/share/racket/collects/db/private/sqlite3/connection.rkt:403:0: handle-status*
raco setup:      /home/alex/tmp/racket/usr/share/racket/collects/db/private/sqlite3/connection.rkt:360:8
raco setup:      /home/alex/tmp/racket/usr/share/racket/collects/db/private/sqlite3/connection.rkt:181:4: step method in connection%
raco setup:      /home/alex/tmp/racket/usr/share/racket/collects/db/private/sqlite3/connection.rkt:169:8: loop
raco setup:      /home/alex/tmp/racket/usr/share/racket/collects/db/private/sqlite3/connection.rkt:72:4: query1 method in connection%
raco setup:      /home/alex/tmp/racket/usr/share/racket/collects/db/private/generic/functions.rkt:165:0: query-rows8
raco setup:      /home/alex/tmp/racket/usr/share/racket/collects/setup/doc-db.rkt:84:4
raco setup:      /home/alex/tmp/racket/usr/share/racket/collects/setup/doc-db.rkt:497:0: call-with-retry/transaction
raco setup:      /home/alex/tmp/racket/usr/share/racket/collects/setup/doc-db.rkt:89:0: doc-db-key->path22
raco setup:      /home/alex/tmp/racket/usr/share/racket/pkgs/racket-index/setup/xref.rkt:121:6: try
raco setup:      /home/alex/tmp/racket/usr/share/racket/pkgs/racket-index/setup/xref.rkt:117:2
raco setup:      /home/alex/tmp/racket/usr/share/racket/pkgs/scribble-lib/scribble/xref.rkt:65:19: send-arg73
raco setup:      /home/alex/tmp/racket/usr/share/racket/pkgs/scribble-lib/scribble/base-render.rkt:1114:28
raco setup:      /home/alex/tmp/racket/usr/share/racket/pkgs/scribble-lib/scribble/base-render.rkt:1114:28
raco setup:      /home/alex/tmp/racket/usr/share/racket/pkgs/scribble-lib/scribble/core.rkt:24:0: resolve-get/where
raco setup:      /home/alex/tmp/racket/usr/share/racket/pkgs/scribble-lib/scribble/core.rkt:73:0: resolve-search
raco setup:      ...
raco setup:   
raco setup: error: during building docs for <pkgs>/draw-doc/scribblings/draw/draw.scrbl
raco setup:   query-rows: attempt to write a readonly database
raco setup:     error code: 776
raco setup:     SQL: "SELECT pathid FROM documented WHERE stag=$1"
raco setup:     database: #<path:/home/alex/tmp/racket/usr/share/doc/racket/docindex.sqlite>
raco setup:     mode: 'read-only
raco setup:     file permissions: (write read)
raco setup:     context...:
raco setup:      /home/alex/tmp/racket/usr/share/racket/collects/db/private/sqlite3/connection.rkt:403:0: handle-status*
raco setup:      /home/alex/tmp/racket/usr/share/racket/collects/db/private/sqlite3/connection.rkt:360:8
raco setup:      /home/alex/tmp/racket/usr/share/racket/collects/db/private/sqlite3/connection.rkt:181:4: step method in connection%
raco setup:      /home/alex/tmp/racket/usr/share/racket/collects/db/private/sqlite3/connection.rkt:169:8: loop
raco setup:      /home/alex/tmp/racket/usr/share/racket/collects/db/private/sqlite3/connection.rkt:72:4: query1 method in connection%
raco setup:      /home/alex/tmp/racket/usr/share/racket/collects/db/private/generic/functions.rkt:165:0: query-rows8
raco setup:      /home/alex/tmp/racket/usr/share/racket/collects/setup/doc-db.rkt:84:4
raco setup:      /home/alex/tmp/racket/usr/share/racket/collects/setup/doc-db.rkt:497:0: call-with-retry/transaction
raco setup:      /home/alex/tmp/racket/usr/share/racket/collects/setup/doc-db.rkt:89:0: doc-db-key->path22
raco setup:      /home/alex/tmp/racket/usr/share/racket/pkgs/racket-index/setup/xref.rkt:121:6: try
raco setup:      /home/alex/tmp/racket/usr/share/racket/pkgs/racket-index/setup/xref.rkt:117:2
raco setup:      /home/alex/tmp/racket/usr/share/racket/pkgs/scribble-lib/scribble/xref.rkt:65:19: send-arg73
raco setup:      /home/alex/tmp/racket/usr/share/racket/pkgs/scribble-lib/scribble/base-render.rkt:1114:28
raco setup:      /home/alex/tmp/racket/usr/share/racket/pkgs/scribble-lib/scribble/base-render.rkt:1114:28
raco setup:      /home/alex/tmp/racket/usr/share/racket/pkgs/scribble-lib/scribble/core.rkt:24:0: resolve-get/where
raco setup:      /home/alex/tmp/racket/usr/share/racket/pkgs/scribble-lib/scribble/core.rkt:73:0: resolve-search
raco setup:      ...
raco setup:   
Makefile:160: recipe for target 'install-3m' failed
make[1]: *** [install-3m] Error 1
make[1]: Leaving directory '/home/alex/dev/nat/racket/racket-6.2.1/src/build'
Makefile:86: recipe for target 'install' failed
```

I was using following commands to build:
```
../configure --build=x86_64-redhat-linux-gnu --host=x86_64-redhat-linux-gnu --program-prefix= --disable-dependency-tracking --prefix=/usr --exec-prefix=/usr --bindir=/usr/bin --sbindir=/usr/sbin --sysconfdir=/etc --datadir=/usr/share --includedir=/usr/include --libdir=/usr/lib64 --libexecdir=/usr/libexec --localstatedir=/var --sharedstatedir=/var/lib --mandir=/usr/share/man --infodir=/usr/share/info
make
make install DESTDIR=/home/alex/tmp/racket
```

**Comment by stamourv** [comment]
That's usually a transient error. Running `make install` again should solve the problem.

**Comment by atorkhov** [comment]
It most likely appears when doing build from scratch? I did builds with deleting build dir twice and twice got an error.

I'm building rpm package of racket. And having a package that sometimes might be built and sometimes not is not very good... :-)

**Comment by stamourv** [comment]
I agree. I think that particular bug was fixed for the (soon to be released) 6.3 version, though.

**Comment by mflatt** [comment]
Yes, v6.3 includes a repair for the "attempt to write a readonly database" bug.
